internal/controllers/machineset/machineset_controller.go

```diff
@@ -278,6 +279,15 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 			r.recorder.Eventf(machineSet, corev1.EventTypeNormal, "SuccessfulAdopt", "Adopted Machine %q", machine.Name)
 		}
 
+		// Ensure that the version of the controllerReference is up-to-date.
+		if metav1.GetControllerOf(machine).APIVersion != machineSet.APIVersion {
```

Comment:
Are sure the machineSet.APIVersion is set? (afaik per default it's not in CR / client-go)

I think we should use the same source of truth as updateControllerReference (which is machineSetKind)